src/prefect/cli/run.py

```diff
@@ -110,6 +114,10 @@ def flow(
         --no-url                                Only output the flow run id instead of a
                                                 link
 
+    \b
+    Either `id` or both `name` and `project` must be provided to run a flow. If all three
+    are provided then `id` will be used when calling the mutation.
```

Comment:
Perhaps instead of using the flow id and ignoring the project & name, providing both should error?